moveit_core/robot_model/include/moveit/robot_model/joint_model_group.h

```diff
@@ -76,7 +76,7 @@ class JointModelGroup
     }
 
     /// Return a flag indicating whether the state of the solver is initialized
-    operator bool() const
+    explicit operator bool() const
```

Comment:
I think implicit conversions like this are useful since you can write `if (joint_mode_group)` instead of `if (bool{joint_model_group})`. Ultimately it's a design decision for you to make whether you want convenience or explicitness, but I think it's no less safe to let these conversions happen quietly.

Reply:
Ok that makes sense, I was just trying to follow the cpp_best_practices in improving the codebase on general standards. Are there scenarios that we want to precisely target, if we don't want an API breakage in conversion operators, but still want implicit conversions at compile time? I don't know if there are defined situations in moveit 2 where it's needed on hand, outside from a search through comments explaining functions.

Reply:
> Are there scenarios that we want to precisely target, if we don't want an API breakage in conversion operators, but still want implicit conversions at compile time?

Can you explain what this means?

Reply:
I assumed when you said
> I agree that explicit is generally better than the lack of it but I'm 100% on board with explicit type conversion operators

You meant that there are some scenarios in this repository where `explicit` is useful to limit implicit conversions from happening and other instances when leaving it as implicit, so don't have to specify, is much easier, essentially it all depends on each use case for the function. I then, wanted to know if I needed to look, in depth, on each single parameter function that could utilize explicit and see if it is worth it to restrict that function. I apologize for miswording.

Reply:
I would recommend against adding `explicit` to these conversion operators.

Reply:
I have since, removed the explicit from all the operators that I have changed, but what about the many single parameter constructors? Most of then I have seen don't have a lot of of changes to be made within the codebase, but they're public so they would also technically break the API with many users.